uxarray/io/_esmf.py

```diff
@@ -73,22 +79,22 @@ def _read_esmf(in_ds):
             "Reading in ESMF grids with Cartesian coordinates not yet supported"
         )
 
-    n_nodes_per_face = in_ds["numElementConn"].values.astype(INT_DTYPE)
+    n_nodes_per_face = in_ds["numElementConn"].astype(INT_DTYPE)
     out_ds["n_nodes_per_face"] = xr.DataArray(
         data=n_nodes_per_face,
         dims=ugrid.N_NODES_PER_FACE_DIMS,
         attrs=ugrid.N_NODES_PER_FACE_ATTRS,
     )
 
-    if "start_index" in in_ds["elementConn"]:
+    if "start_index" in in_ds["elementConn"].attrs:
         start_index = in_ds["elementConn"].start_index
     else:
         # assume start index is 1 if one is not provided
         start_index = 1
 
     face_node_connectivity = in_ds["elementConn"].values.astype(INT_DTYPE)
 
-    for i, max_nodes in enumerate(n_nodes_per_face):
+    for i, max_nodes in enumerate(n_nodes_per_face.values):
```

Comment:
I think this is needed after all the above eliminations of `.values`, right? Could it still be handled without explicit conversion here? In case of Dask, isn't `n_nodes_per_face` be a Dask array, too?

I am wondering if `.data` instead of `.values` could effectively be used in such cases. In that case, `.data` would either return numpy arrays or Dask arrays, and maybe both cases would need to be handled separately.

Reply:
Good catch! 

I didn't remove the `.values()` call above. Let me see if I can update it to not require it.

Reply:
I've replaced it with the following, which is what I used for MPAS

```Python
face_node_connectivity = in_ds["elementConn"].astype(INT_DTYPE)
face_node_connectivity = xr.where(face_node_connectivity != INT_FILL_VALUE, face_node_connectivity - start_index, face_node_connectivity)
```

Reply:
Thanks!